src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/VariantOfListFW.java

```diff
@@ -0,0 +1,302 @@
+            list32RO.wrap(buffer, offset + enumWithInt8.sizeof(), maxLimit);
+            break;
+        case TWO:
+            list8RO.wrap(buffer, offset + enumWithInt8.sizeof(), maxLimit);
+            break;
+        case THREE:
+            list0RO.wrap(buffer, offset + enumWithInt8.sizeof(), maxLimit);
+            break;
+        default:
+            break;
+        }
+        checkLimit(limit(), maxLimit);
+        return this;
+    }
+
+    @Override
+    public VariantOfListFW tryWrap(
+        DirectBuffer buffer,
+        int offset,
+        int maxLimit)
+    {
+        if (super.tryWrap(buffer, offset, maxLimit) == null)
+        {
+            return null;
+        }
+        Flyweight enumWithInt8 = enumWithInt8RO.tryWrap(buffer, offset, maxLimit);
+        if (enumWithInt8 == null)
+        {
+            return null;
+        }
+        switch (kind())
+        {
+        case ONE:
+            if (list32RO.tryWrap(buffer, offset + enumWithInt8.sizeof(), maxLimit) == null)
+            {
+                return null;
+            }
+            break;
+        case TWO:
+            if (list8RO.tryWrap(buffer, offset + enumWithInt8.sizeof(), maxLimit) == null)
+            {
+                return null;
+            }
+            break;
+        case THREE:
+            if (list0RO.tryWrap(buffer, offset + enumWithInt8.sizeof(), maxLimit) == null)
+            {
+                return null;
+            }
+            break;
+        default:
+            break;
+        }
+        if (limit() > maxLimit)
+        {
+            return null;
+        }
+        return this;
+    }
+
+    @Override
+    public int limit()
+    {
+        return get().limit();
+    }
+
+    public static final class Builder extends Flyweight.Builder<VariantOfListFW>
+    {
+        private final List32FW.Builder list32RW = new List32FW.Builder();
+
+        private final List8FW.Builder list8RW = new List8FW.Builder();
+
+        private final List0FW.Builder list0FW = new List0FW.Builder();
+
+        private final EnumWithInt8FW.Builder enumWithInt8RW = new EnumWithInt8FW.Builder();
+
+        public Builder()
+        {
+            super(new VariantOfListFW());
+        }
+
+        private Builder kind(
+            EnumWithInt8 value)
+        {
+            enumWithInt8RW.wrap(buffer(), offset(), maxLimit());
+            enumWithInt8RW.set(value);
+            limit(enumWithInt8RW.build().limit());
+            return this;
+        }
+
+        public Builder field(
+            Flyweight.Builder.Visitor mutator)
+        {
+            list32RW.field(mutator);
+            limit(list32RW.limit());
+            return this;
+        }
+
+        public Builder setAsList32(
+            ListFW list)
+        {
+            kind(KIND_ONE);
+            List32FW.Builder list32 = list32RW.wrap(buffer(), limit(), maxLimit());
+            list32.fields(list.fieldCount(), list.fields(), 0, list.length() - BitUtil.SIZE_OF_INT);
```

Comment:
Should last parameter value be `list.fields().capacity()` instead of `list.length() - BitUtil.SIZE_OF_INT`?

Reply:
Oh yes that is right!